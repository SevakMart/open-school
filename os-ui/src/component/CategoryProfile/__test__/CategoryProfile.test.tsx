import { render, screen } from '@testing-library/react';
import CategoryCard from '../CategoryProfile';

const categoryProfile = {
  title: 'JavaScript',
  logoPath: 'https://reactjs.org/logo-og.png',
  subCategoryCount: 3,

};
describe('Create unit tests of category profile', () => {
  test('Make a snapshot tests', () => {
    const { asFragment } = render(<CategoryCard category={categoryProfile} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Verify if category info is contained in the component', () => {
    render(<CategoryCard category={categoryProfile} />);
    const titleElem = screen.queryByTestId('JavaScript');
    expect(titleElem).toBeInTheDocument();
    expect(titleElem).toHaveTextContent('JavaScript');
  });
});
