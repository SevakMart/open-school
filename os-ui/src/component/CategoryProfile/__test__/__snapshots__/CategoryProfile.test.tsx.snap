// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create unit tests of category profile Make a snapshot tests 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <img
      alt="JavaScript"
      src="CategoryIcon.svg"
    />
    <div
      class="categoryInfoContainer"
    >
      <div>
        <p
          class="categoryTitle"
          data-testid="JavaScript"
        >
          JavaScript
        </p>
        <p
          class="numberOfSubcategories"
        >
          3
        </p>
      </div>
      <img
        alt="JavaScript"
        src="RightArrow.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
